service/service.go

```diff
@@ -397,8 +397,20 @@ type RunningDriver struct {
 }
 
 func (r *RunningDriver) Start() (string, error) {
-	port := rand.Intn(1000) + 10*1000 // KontainerDriver port range 10,000 - 11,000
-	listenAddress := fmt.Sprintf("%v%v", ListenAddress, port)
+	ephemeralListenAddress := fmt.Sprintf("%s:0", ListenAddress)
+	p, err := net.Listen("tcp", ephemeralListenAddress)
+	if err != nil {
+		return "", err
+	}
+
+	listenAddress := p.Addr().String()
+	p.Close()
+
+	portString := portOnly(listenAddress)
+	port, err := strconv.Atoi(portString)
```

Comment:
nitpick, but this could be part of the `portOnly` method so any future use would get an error also.